packages/react/src/components/header/components/headerActionBar/HeaderActionBarNavigationMenu.tsx

```diff
@@ -1,47 +1,348 @@
+  onDropdownButtonClick: (link) => void;
+  onLinkClick: (link) => void;
+};
+const MenuLinks = ({ links, onDropdownButtonClick, onLinkClick }: MenuLinksProps) => {
+  return (
+    <>
+      {links.map((child, index) => {
+        if (!isValidElement(child)) return null;
+        return (
+          // eslint-disable-next-line react/no-array-index-key
+          <li key={index}>
+            <span className={styles.mobileNavigationLink}>
+              {cloneElement(child as React.ReactElement, {
+                dropdownLinkClassName: styles.mobileDropdownLink,
+                dropdownClassName: styles.mobileLinkDropdown,
+                wrapperClassName: styles.mobileLinkWrapper,
+                dropdownButtonClassName: styles.mobileLinkDropdownButton,
+                className: classNames(child.props.className, styles.mobileLink),
+                index,
+                onDropdownButtonClick: () => onDropdownButtonClick(child),
+                onClick: () => onLinkClick(child),
+              })}
+            </span>
+          </li>
+        );
+      })}
+    </>
+  );
+};
+
+/**
+ * Find only the active links in the given objects model.
+ * @param obj
+ * @returns
+ */
+function findActiveLinks(obj) {
+  const activeLinks = [];
+
+  if (obj.props.active) {
+    activeLinks.push(obj);
+  }
+
+  if (obj.props.dropdownLinks && obj.props.dropdownLinks.length > 0) {
+    obj.props.dropdownLinks.forEach((link) => {
+      activeLinks.push(...findActiveLinks(link));
+    });
+  }
+
+  return activeLinks;
+}
+
+function getActiveMainLevelLink(links) {
+  return getChildrenAsArray(links).find((link) => {
+    if (isValidElement(link) && link.props.active) return link;
+    return null;
+  });
+}
+
+// These are used as clasNames as well
+type Position = 'left0' | 'left100' | 'left200';
+type HeaderActionBarNavigationMenuProps = {
+  frontPageLabel: string;
+  titleHref: string;
+};
+export const HeaderActionBarNavigationMenu = ({ frontPageLabel, titleHref }: HeaderActionBarNavigationMenuProps) => {
+  const { navigationContent, mobileMenuOpen, hasUniversalContent, universalContent } = useHeaderContext();
+  const { setMobileMenuOpen } = useSetHeaderContext();
+  // State for which link menu is open but not necessarily active. Needed for browsing the menu.
+  const [openMainLinks, setOpenMainLinks] = useState<React.ReactElement[]>([]);
+  // State for the link with dropdowns that the user is opening. Needed for rendering next menu and to show its data while animating.
+  const [openingLink, setOpeningLink] = useState<React.ReactElement | string>(null);
+  // State for the wide wrapping element's position. Value is also used as a class for animation.
+  const [position, setPosition] = useState<Position>('left0');
+  const [isAnimating, setIsAnimating] = useState<boolean>(false);
+  const navContainerRef = useRef<HTMLDivElement>();
+  const isOpeningLinkFromBefore = (link) => !!openMainLinks[openMainLinks.indexOf(link)];
```

Comment:
I find this function a bit out of place hanging around here with the constants, I would suggest moving it below all the others (grouping constants with other constants and functions with functions etc.) 🙂 The parameter `link` is type also of any 🤔